### PR TITLE
Upgrade to JUnit Platform 5.8

### LINKED ISSUE #131: Upgrade to JUnit 5.8
## Overview

JUnit 5.8 has been released with the official `UniqueIdTrackingListener`.

## Deliverables

- [x] Upgrade the build to use JUnit 5.8.0
- [x] Update the Gradle and Maven plugins to use the official `UniqueIdTrackingListener` (requires setting a system property to enable the listener)
- [x] Decide whether the current "local" version of the `UniqueIdTrackingListener` should be removed. If it's not removed, make sure that only one version of the `UniqueIdTrackingListener` is used per test suite

### PR DESCRIPTION
**Pull request by melix**
Closes #131

### CODE REVIEW

File: docs/src/docs/asciidoc/index.adoc [hunk 1]
```diff
@@ -15,6 +15,28 @@ If you are interested in contributing, please refer to our https://github.com/gr
 [[changelog]]
 == Changelog
 
+=== Release 0.9.6
+
+==== Upgrade to JUnit 5.8
+
+The plugins now depend on JUnit 5.8 which provides official an official test listener which is used by these plugins.
```
Comment (ttddyy) [inline docs/src/docs/asciidoc/index.adoc:22]: nit typo: "...provides official an official test ..."

File: native-maven-plugin/src/seeding-build/pom.xml [hunk 2]
```diff
@@ -51,7 +51,7 @@
     <properties>
         <java.version>1.8</java.version>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <junit.jupiter.version>5.7.2</junit.jupiter.version>
+        <junit.jupiter.version>5.8.0</junit.jupiter.version>
```
Comment (ttddyy) [inline native-maven-plugin/src/seeding-build/pom.xml:54]: 5.8.0 -> 5.8.1 ??

Comment (melix) [reply native-maven-plugin/src/seeding-build/pom.xml:54]: Good catch!

**Comment by melix** [comment]
@sbrannen would you have time to review this?

**Comment by sbrannen** [comment]
> @sbrannen would you have time to review this?

I'll see if I can squeeze in a review before the end of the week

File: native-gradle-plugin/src/main/java/org/graalvm/buildtools/gradle/NativeImagePlugin.java [hunk 3]
```diff
@@ -131,6 +131,7 @@
      */
     private static final Consumer<Object> FORCE_CONFIG = t -> {
     };
+    private static final String JUNIT_PLATFORM_LISTENER_ENABLE = "junit.platform.listeners.uid.tracking.enabled";
```
Comment (sbrannen) [inline native-gradle-plugin/src/main/java/org/graalvm/buildtools/gradle/NativeImagePlugin.java:134]:
```suggestion
    private static final String JUNIT_PLATFORM_LISTENERS_UID_TRACKING_ENABLED = "junit.platform.listeners.uid.tracking.enabled";
```

For consistency with `NativeExtension`.

**Review by sbrannen — APPROVED** [review]
I think this PR looks good. 👍 

I only noticed one minor inconsistency regarding a constant name.

**Review by graemerocher — APPROVED** [review]

**Pull request merged**